### PR TITLE
Adp 2113 test logger

### PR DESCRIPTION
**Pull request by iccicci**
# Context

Sometimes it is hard to investigate unit tests failure because we often use `dummyLogger`.
To investigate problem we often need to edit the test file to substitute `dummyLogger` with a real logger.
This makes the test output _not clean_, so at the end of the investigation it is a good practice to restore `dummyLogger`...
This process is often a heavy time consumer.

# Proposed Solution

- Added `TestLogger`, a simple unit tests dedicated logger.
- Used `TestLogger` in `@cardano-sdk/rabbitmq` as the latter uses the `record` feature of the new logger.
- Used `TestLogger` in `@cardano-sdk/cardano-services-client` to produce a _clean_ test output.

### CODE REVIEW

File: packages/util-dev/src/TestLogger.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * Options for createLogger
+ */
+export interface TestLoggerOptions {
+  /**
+   * The environment variables to use. Default: process.env - Used to test TestLogger
+   */
+  env?: NodeJS.ProcessEnv;
+
+  /**
+   * If true, the logger will records all the logged values
```
Comment (mirceahasegan) [inline packages/util-dev/src/TestLogger.ts:47]: Is the default 'false'?

File: packages/util-dev/test/TestLogger.test.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ describe('TestLogger', () => {
     });
 
     describe('TL_BREAK', () => {
-      const loggedData = Object.fromEntries(Array.from({ length: 10 }, (v, i) => [i.toString(), `v${i}`]));
+      const loggedData = Object.fromEntries(Array.from({ length: 10 }, (_v, i) => [i.toString(), `v${i}`]));
```
Comment (mirceahasegan) [inline packages/util-dev/test/TestLogger.test.ts:90]: Why `_v` here?

Comment (iccicci) [reply packages/util-dev/test/TestLogger.test.ts:90]: Because it is unused. Without the underscore `eslint` raises an error (or warning...)

Comment (mirceahasegan) [reply packages/util-dev/test/TestLogger.test.ts:90]: Oh. I see. Didn't notice it's not used. I've seen the following conventions until now.
```
(_, i) => [i.toString(), `v${i}`];
// or
(, i) => [i.toString(), `v${i}`];

Comment (IvayloAndonov) [reply packages/util-dev/test/TestLogger.test.ts:90]: I'm using that one `(_, i) => [i.toString(), `v${i}`];`

File: packages/util-dev/src/TestLogger.ts (hunk 1 above)
Comment (iccicci) [reply packages/util-dev/src/TestLogger.ts]: b6849e8a6 should make the question disappear.

File: packages/util-dev/test/TestLogger.test.ts (hunk 2 above)
Comment (iccicci) [reply packages/util-dev/test/TestLogger.test.ts]: b6849e8a6 should solve this

**Comment by mirceahasegan** [comment]
Nice work @iccicci! 🎉

**Review by mirceahasegan — APPROVED** [review]

**Review by AngelCastilloB — DISMISSED** [review]
LGTM Awesome job @iccicci ! (I really liked the explanation on the README file :)

**Review by mkazlauskas — DISMISSED** [review]
Nice work @iccicci, good to squash :rocket:

**Review by IvayloAndonov — APPROVED** [review]
Nice work!

**Review by mirceahasegan — DISMISSED** [review]

**Review by mkazlauskas — DISMISSED** [review]

**Review by mirceahasegan — APPROVED** [review]

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**